store/snapshot/store.go

```diff
@@ -0,0 +1,354 @@
+package snapshot
```

Comment:
I agree that this probably doesn't belong under `store/`. I would reason that I should probably be its own root package (it's not a module either). In addition, I'd also move the snapshot types you have in `types/` to this new root package as well.

Reply:
>I'd also move the snapshot types you have in types/ to this new root package as well.

Well, it depends on how we structure this. I was thinking the snapshot store shouldn't care about what the snapshots actually contain, they're just opaque binary blobs. The actual snapshotting is the responsibility of the data store (i.e. `rootmulti.Store`). This decouples them cleanly, such that the snapshot store doesn't have to know about the data store's internals. Thus the `SnapshotItem` and related types (snapshot contents) go with `rootmulti.Store`, while `SnapshotMetadata` and `SnapshotChunkMetadata` go with `snapshot.Store`.

Reply:
Makes sense! But I'd still argue that they don't belong in `types/`. If they are specific to how the root multi-store behaves, shouldn't they be defined in scope of the root multi-store (i.e. the store pkg).

Reply:
Yes, you're right of course. I added them in the root types because I didn't see any Protobufs for the `store` types, and forgot about it. I'll reorganize it a bit.